graphene_mongo/utils.py

```diff
@@ -101,3 +102,61 @@ def get_node_from_global_id(node, info, global_id):
                 return interface.get_node_from_global_id(info, global_id)
     except AttributeError:
         return Node.get_node_from_global_id(info, global_id)
+
+
+def collect_query_fields(node, fragments):
+    """Recursively collects fields from the AST
+
+    Args:
+        node (dict): A node in the AST
+        fragments (dict): Fragment definitions
+
+    Returns:
+        A dict mapping each field found, along with their sub fields.
+
+        {'name': {},
+         'image': {'id': {},
+                                   'name': {},
+                                   'description': {}},
```

Comment:
The indent is a bit weird here, would you please re-indent here?

Reply:
Done